src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -1349,6 +1350,18 @@ def build_vmss_resource(cmd, name, computer_name_prefix, location, tags, overpro
         }
         virtual_machine_profile['scheduledEventsProfile'] = scheduled_events_profile
 
+    if enable_osimage_notification is not None:
+        os_image_notification_profile = {
+            'osImageNotificationProfile': {
+                'enable': enable_osimage_notification
+            }
+        }
+        if terminate_notification_time is not None:
+            scheduled_events_profile.update(os_image_notification_profile)
+            virtual_machine_profile['scheduledEventsProfile'] = scheduled_events_profile
```

Comment:
>  if terminate_notification_time is not None

Why is `if terminate_notification_time is not None` instead of `if scheduled_events_profile is not None` ?